be/src/olap/rowset/segment_v2/page_compression.cpp

```diff
@@ -0,0 +1,88 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "olap/rowset/segment_v2/page_compression.h"
+
+#include "gutil/strings/substitute.h"
+#include "util/block_compression.h"
+#include "util/coding.h"
+
+namespace doris {
+namespace segment_v2 {
+
+using strings::Substitute;
+
+Status PageDecompressor::init() {
+    if (_data.size < 4) {
+        return Status::Corruption(
+            Substitute("Compressed page's size is too small, size=$0, needed=$1",
+                       _data.size, 4));
+    }
+    _uncompressed_bytes = decode_fixed32_le((uint8_t*)_data.data);
+    return Status::OK();
+}
+
+Status PageDecompressor::decompress_to(void* buf) {
+    Slice compressed_slice(_data.data + 4, _data.size - 4);
+    if (compressed_slice.size == _uncompressed_bytes) {
```

Comment:
Actually, for this case, we will only store uncompressed data. 
In PageCompressor, if compressed size >= uncompressed size we will store uncompressed data.
I prefer this method, because this method is easy to parse and use less space.

Reply:
Yeah, you're right